File: packages/compiler/shared/diagnostics.ts
// eslint-disable-next-line no-shadow
export const enum DiagnosticCode {
  ERROR = 1000,
  ERROR_UNTERMINATED_JS_COMMENT = 1001,
  ERROR_FRAGMENT_SHORTHAND_ATTRS = 1002,
  ERROR_UNMATCHED_IMPORT = 1003,
  ERROR_UNSUPPORTED_SLOT_ATTRIBUTE = 1004,
  WARNING = 2000,
  WARNING_UNTERMINATED_HTML_COMMENT = 2001,
  WARNING_UNCLOSED_HTML_TAG = 2002,
  WARNING_DEPRECATED_DIRECTIVE = 2003,
  WARNING_IGNORED_DIRECTIVE = 2004,
  WARNING_UNSUPPORTED_EXPRESSION = 2005,
  WARNING_SET_WITH_CHILDREN = 2006,
  INFO = 3000,
  HINT = 4000,
}

File: internal/loc/diagnostics.go
package loc

type DiagnosticCode int

const (
	ERROR                             DiagnosticCode = 1000
	ERROR_UNTERMINATED_JS_COMMENT     DiagnosticCode = 1001
	ERROR_FRAGMENT_SHORTHAND_ATTRS    DiagnosticCode = 1002
	ERROR_UNMATCHED_IMPORT            DiagnosticCode = 1003
	ERROR_UNSUPPORTED_SLOT_ATTRIBUTE  DiagnosticCode = 1004
	ERROR_UNTERMINATED_STRING         DiagnosticCode = 1005
	WARNING                           DiagnosticCode = 2000
	WARNING_UNTERMINATED_HTML_COMMENT DiagnosticCode = 2001
	WARNING_UNCLOSED_HTML_TAG         DiagnosticCode = 2002
	WARNING_DEPRECATED_DIRECTIVE      DiagnosticCode = 2003
	WARNING_IGNORED_DIRECTIVE         DiagnosticCode = 2004
	WARNING_UNSUPPORTED_EXPRESSION    DiagnosticCode = 2005
	WARNING_SET_WITH_CHILDREN         DiagnosticCode = 2006
	WARNING_CANNOT_DEFINE_VARS        DiagnosticCode = 2007
	WARNING_INVALID_SPREAD            DiagnosticCode = 2008
	WARNING_UNEXPECTED_CHARACTER      DiagnosticCode = 2009
	WARNING_CANNOT_RERUN              DiagnosticCode = 2010
	INFO                              DiagnosticCode = 3000
	HINT                              DiagnosticCode = 4000
)
